### PR TITLE
Add `print_verbose()` built-in function to print in verbose mode only

### LINKED ISSUE #3324: Expose `print_verbose()` to the scripting API
### Describe the project you are working on

The Godot editor :slightly_smiling_face: 

### Describe the problem or limitation you are having in your project

Godot core features a `print_verbose()` method that works like its `print_line()` method, but only when verbose mode is enabled (`OS.is_verbose_stdout()` returning `true`).

However, this feature is not exposed to the scripting API.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Expose `print_verbose()` to the scripting API as a convenience feature.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

See the implementation here: https://github.com/godotengine/godot/pull/52877

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Yes:

```gdscript
if OS.is_stdout_verbose():
    print("...")
```

However, consider that printing verbose statements is something that will be done often in more complex scripts. Having to add this `if` statement quickly adds up over time.

### Is there a reason why this should be core and not an add-on in the asset library?

See above.

### PR DESCRIPTION
**Pull request by Calinou**
`master` version of https://github.com/godotengine/godot/pull/53590.

This can be used as a shorthand for:

```gdscript
if OS.is_stdout_verbose():
    print("...")
```

Unlike `print_debug()`, this works in release builds too and can be toggled off in debug builds.

This closes https://github.com/godotengine/godot-proposals/issues/3324.

## Preview

```gdscript
extends Node

func _ready():
	print("normal", " and ", "verbose")
	print_verbose("verbose", " only")
```

### Verbose mode disabled

```
normal and verbose
```

### Verbose mode enabled

*Enable verbose stdout in the advanced project settings:*

image

*Alternatively, you can start Godot with the `--verbose`/`-v` command line argument.*

```
normal and verbose
verbose only
```

### CODE REVIEW

**Comment by fire** [comment]
Weird error on Linux.

**Comment by Calinou** [comment]
> Weird error on Linux.

This is because the built-in function enum was updated, so the class reference needs to be regenerated.

I don't know if I should add the new built-in function at the end for the `master` version of this PR.

File: core/variant/variant_utility.cpp [hunk 1]
```diff
@@ -506,6 +506,33 @@ struct VariantUtilityFunctions {
 		r_error.error = Callable::CallError::CALL_OK;
 	}
 
+	static inline void print_verbose(const Variant **p_args, int p_arg_count, Callable::CallError &r_error) {
+		if (p_arg_count < 1) {
+			r_error.error = Callable::CallError::CALL_ERROR_TOO_FEW_ARGUMENTS;
+			r_error.argument = 1;
+		}
```
Comment (vnen) [inline core/variant/variant_utility.cpp:513]: Shouldn't this return if an error is found?

Comment (vnen) [reply core/variant/variant_utility.cpp:513]: Now that think of it, there's not really a reason to give an error when no arguments are provided. `print()` can be used to print a single newline.

Comment (Calinou) [reply core/variant/variant_utility.cpp:513]: I agree with that too – in fact, it already seems to work that way in GDScript.

**Review by vnen — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**